Schema Registry client does not support security configs

When the Schema Registry client is built here https://github.com/purbon/kafka-topology-builder/blob/e97fc7390cd7e60bfd0e481b1916bf71815f1e67/src/main/java/com/purbon/kafka/topology/KafkaTopologyBuilder.java#L73, it uses no config constructor. This means we can not pass security parameters. 

Can we pass parameters from the tool config (maybe prefixed?) here? and use this constructor:

https://github.com/confluentinc/schema-registry/blob/master/client/src/main/java/io/confluent/kafka/schemaregistry/client/CachedSchemaRegistryClient.java#L106

Ok.  So I created brand new Schema Registry API Key and another same for kafka cluster (based on my user).   This is what I got with that image (no log4j setup I'm afraid:

2020-09-29T13:34:37.1690295Z *** show contents of current directory for debugging ***
2020-09-29T13:34:37.1693299Z README.md
2020-09-29T13:34:37.1693913Z deploy-topology.yml
2020-09-29T13:34:37.1694063Z kafka_topology
2020-09-29T13:34:37.1694197Z poc_telemetry
2020-09-29T13:34:37.1694495Z simulate-topology-deployment.yml
2020-09-29T13:34:37.1694812Z topology-builder.properties
2020-09-29T13:34:37.6824490Z log4j:WARN No appenders could be found for logger (org.apache.kafka.clients.admin.AdminClientConfig).
2020-09-29T13:34:37.6824919Z log4j:WARN Please initialize the log4j system properly.
2020-09-29T13:34:37.6825249Z log4j:WARN See http://logging.apache.org/log4j/1.2/faq.html#noconfig for more info.
2020-09-29T13:34:39.2297033Z Exception in thread "main" com.purbon.kafka.topology.schemas.SchemaRegistryManager$SchemaRegistryManagerException: Failed to parse the schema file schemas/bar-key.avsc
2020-09-29T13:34:39.2304614Z 	at com.purbon.kafka.topology.schemas.SchemaRegistryManager.register(SchemaRegistryManager.java:45)
2020-09-29T13:34:39.2342018Z 	at com.purbon.kafka.topology.actions.topics.SyncTopicAction.syncTopic(SyncTopicAction.java:60)
2020-09-29T13:34:39.2361510Z 	at com.purbon.kafka.topology.actions.topics.SyncTopicAction.run(SyncTopicAction.java:41)
2020-09-29T13:34:39.2361953Z 	at com.purbon.kafka.topology.ExecutionPlan.execute(ExecutionPlan.java:66)
2020-09-29T13:34:39.2362302Z 	at com.purbon.kafka.topology.ExecutionPlan.run(ExecutionPlan.java:50)
2020-09-29T13:34:39.2362667Z 	at com.purbon.kafka.topology.KafkaTopologyBuilder.run(KafkaTopologyBuilder.java:105)
2020-09-29T13:34:39.2363059Z 	at com.purbon.kafka.topology.KafkaTopologyBuilder.run(KafkaTopologyBuilder.java:116)
2020-09-29T13:34:39.2363672Z 	at com.purbon.kafka.topology.BuilderCLI.processTopology(BuilderCLI.java:178)
2020-09-29T13:34:39.2364017Z 	at com.purbon.kafka.topology.BuilderCLI.run(BuilderCLI.java:131)
2020-09-29T13:34:39.2364341Z 	at com.purbon.kafka.topology.BuilderCLI.main(BuilderCLI.java:120)
2020-09-29T13:34:39.2365382Z Caused by: com.purbon.kafka.topology.schemas.SchemaRegistryManager$SchemaRegistryManagerException: Failed to parse the schema file poc_telemetry/poc_telemetry/schemas/bar-key.avsc
2020-09-29T13:34:39.2365946Z 	at com.purbon.kafka.topology.schemas.SchemaRegistryManager.register(SchemaRegistryManager.java:54)
2020-09-29T13:34:39.2366405Z 	at com.purbon.kafka.topology.schemas.SchemaRegistryManager.register(SchemaRegistryManager.java:43)
2020-09-29T13:34:39.2366694Z 	... 9 more
2020-09-29T13:34:39.2367388Z Caused by: com.purbon.kafka.topology.schemas.SchemaRegistryManager$SchemaRegistryManagerException: Failed to register the schema for subject 'tbc.azure.comms.tbc.poc-telemetry.bar.avro' of type 'AVRO'
2020-09-29T13:34:39.2367994Z 	at com.purbon.kafka.topology.schemas.SchemaRegistryManager.register(SchemaRegistryManager.java:86)
2020-09-29T13:34:39.2368435Z 	at com.purbon.kafka.topology.schemas.SchemaRegistryManager.register(SchemaRegistryManager.java:52)
2020-09-29T13:34:39.2368717Z 	... 10 more
2020-09-29T13:34:39.2369049Z Caused by: io.confluent.kafka.schemaregistry.client.rest.exceptions.RestClientException: Unauthorized; error code: 401
2020-09-29T13:34:39.2369650Z 	at io.confluent.kafka.schemaregistry.client.rest.RestService.sendHttpRequest(RestService.java:293)
2020-09-29T13:34:39.2370082Z 	at io.confluent.kafka.schemaregistry.client.rest.RestService.httpRequest(RestService.java:352)
2020-09-29T13:34:39.2370528Z 	at io.confluent.kafka.schemaregistry.client.rest.RestService.registerSchema(RestService.java:495)
2020-09-29T13:34:39.2370964Z 	at io.confluent.kafka.schemaregistry.client.rest.RestService.registerSchema(RestService.java:486)
2020-09-29T13:34:39.2371402Z 	at io.confluent.kafka.schemaregistry.client.rest.RestService.registerSchema(RestService.java:459)
2020-09-29T13:34:39.2371897Z 	at io.confluent.kafka.schemaregistry.client.CachedSchemaRegistryClient.registerAndGetId(CachedSchemaRegistryClient.java:206)
2020-09-29T13:34:39.2372414Z 	at io.confluent.kafka.schemaregistry.client.CachedSchemaRegistryClient.register(CachedSchemaRegistryClient.java:268)
2020-09-29T13:34:39.2372919Z 	at io.confluent.kafka.schemaregistry.client.CachedSchemaRegistryClient.register(CachedSchemaRegistryClient.java:244)
2020-09-29T13:34:39.2373400Z 	at com.purbon.kafka.topology.schemas.SchemaRegistryManager.register(SchemaRegistryManager.java:80)
2020-09-29T13:34:39.2373681Z 	... 11 more

Note:  the topic tbc.azure.comms.tbc.poc-telemetry.bar.avro **was** created.

I guess your code change didnt work - or maybe I'm not setting up the parameters in the way that's expected?

I actually have the schema registry credentials in two places:

1. In my devops pipeline I create a new image based on yours, copy a .properties file to it and then run the shell script to deploy from that image.
2. They are also in the topology.yml  file - I dont know which way it should be done.

Snips below to help:

### .properties (redacted)

*** temporarily show contents of topology-builder.properties ***

 bootstrap.servers=pkc-41mxj.uksouth.azure.confluent.cloud:9092 
 security.protocol=SASL_SSL 
 sasl.jaas.config=org.apache.kafka.common.security.plain.PlainLoginModule required username="QDRPE4BIHKMI72KP" password="----"; 
 ssl.endpoint.identification.algorithm=https 
 sasl.mechanism=PLAIN 
 # Confluent Cloud Schema Registry 
 basic.auth.credentials.source=USER_INFO 
 schema.registry.basic.auth.user.info=IZCKNJV2TL3JE7YG:----- 
 confluent.schema.registry.url=https://psrc-lg26v.westeurope.azure.confluent.cloud

 ### topology.yaml redacted

platform:
  schema_registry:
    instances:
      - principal: "IZCKNJV2TL3JE7YG:-----"
#        topic: "foo"
#        group: "bar"
#      - principal: "User:SchemaRegistry02"
#        topic: "zet"

Hi @nikkh , sorry I could not help out earlier, today was a really busy day ... well like all of them 😄 

I pushed a generalisation of the property fetch code in the PR #92, and tested it finally properly with a schema registry in confluent cloud.

Your error says:

```
2020-09-29T13:34:39.2369049Z Caused by: io.confluent.kafka.schemaregistry.client.rest.exceptions.RestClientException: Unauthorized; error code: 401
```
this is because the code was not fetching properly the key and secret for the cloud schema registry instance.

I updated as well the image with the latest code from the PR at:

```
docker pull purbon/kafka-topology-builder:srlatest
```

so you can run latest with the changes I made in #92 .

with that in mind, I tested it and with config like this:

```java
bootstrap.servers={redacted}
security.protocol=SASL_SSL
sasl.jaas.config={redacted}
ssl.endpoint.identification.algorithm=https
sasl.mechanism=PLAIN
# Confluent Cloud Schema Registry
schema.registry.url={redacted}
basic.auth.credentials.source=USER_INFO
schema.registry.basic.auth.user.info={redacted}
```

and a topology with that content as topics

```yaml
  - name: "projectB"
    topics:
      - dataType: "avro"
        name: "bar"
        schemas:
          key.schema.file: "schemas/bar-key.avsc"
          value.schema.file: "schemas/bar-value.avsc"
```

and basic avsc like

```json
{"type": "string"}
```

all process run good.

can you try again and verify I did all proper tests?

once you confirm all is working I will merge #92 and update the latest image from docker hub.

I hope that helps,

Thanks for the quick response - unfortunately I'm still unable to get it to work.  

This is what I did:

1. switched my topology.yaml for yours
2. reconciled the names of all the parameters for the properties file
3. ensured I am pulling the latest image (I note you used the **same name** for this one as the previous attempt)
4. ran my pipeline

First time through it failed - but because our cluster has a policy that the number of replicas must be 3 - but that shows it had authenticated using the sasl.jaas.config entry, and this had permissions to create the topic.

I then edited my topology file to include a specification for number of replicas - which then avoided the policy problem and the newly named topic was created:

`ccloud kafka topic list
                           Name
+---------------------------------------------------------+
 ...
  context.company.env.source.projectB.bar.avro`

but I still got an error:

`com.purbon.kafka.topology.schemas.SchemaRegistryManager.register(SchemaRegistryManager.java:52)
2020-09-29T15:59:38.9963113Z 	... 10 more
2020-09-29T15:59:38.9963460Z Caused by: io.confluent.kafka.schemaregistry.client.rest.exceptions.RestClientException: Unauthorized; error code: 401`

So, in order to test the credentials that I am using in my .properties file, I used CLI to logon with the same API_KEY/SECRET:

`
ccloud schema-registry schema describe 100006
Enter your Schema Registry API key: Enter your Schema Registry API secret:
Failed to validate Schema Registry API key and secret.
Enter your Schema Registry API key: 4LDU...
Enter your Schema Registry API secret: ****************************************************************
Schema: {"type":"record","name":"acceleration","namespace":"colne_valley_viaduct.acceleration","fields":[{"name":"SOSAobservedProperty","type":"string","doc":"https://www.w3.org/TR/vocab-ssn/#SOSAobservedProperty","default":"SOSAobservedProperty"},...
`
I also used the same credentials above (and the ones I am specifying in the .properties file) to manually create a schema and assign it to the topic that was created above:

`ccloud schema-registry schema create --subject context.company.env.source.projectB.bar.avro --type AVRO --schema tempschema.s
Successfully registered schema with ID 100010
ccloud schema-registry schema describe 100010
Schema: "string"
`
For completeness, a redacted version of my properties file is:

`bootstrap.servers=pkc-41mxj.uksouth.azure.confluent.cloud:9092 
 security.protocol=SASL_SSL 
 sasl.jaas.config=org.apache.kafka.common.security.plain.PlainLoginModule required username="KQQP..." password="..."; 
 ssl.endpoint.identification.algorithm=https 
 sasl.mechanism=PLAIN 
 rem Confluent Cloud Schema Registry 
 basic.auth.credentials.source=USER_INFO 
 schema.registry.basic.auth.user.info=4LDU...:... 
 schema.registry.url=https://psrc-lg26v.westeurope.azure.confluent.cloud`


My current log is below:

`*** show contents of current directory for debugging ***
2020-09-29T15:59:37.1687318Z README.md
2020-09-29T15:59:37.1687996Z deploy-topology.yml
2020-09-29T15:59:37.1688278Z kafka_topology
2020-09-29T15:59:37.1688530Z poc_telemetry
2020-09-29T15:59:37.1688961Z simulate-topology-deployment.yml
2020-09-29T15:59:37.1689388Z topology-builder.properties
2020-09-29T15:59:37.6617948Z log4j:WARN No appenders could be found for logger (org.apache.kafka.clients.admin.AdminClientConfig).
2020-09-29T15:59:37.6618665Z log4j:WARN Please initialize the log4j system properly.
2020-09-29T15:59:37.6619122Z log4j:WARN See http://logging.apache.org/log4j/1.2/faq.html#noconfig for more info.
2020-09-29T15:59:38.9919232Z Exception in thread "main" com.purbon.kafka.topology.schemas.SchemaRegistryManager$SchemaRegistryManagerException: Failed to parse the schema file schemas/bar-key.avsc
2020-09-29T15:59:38.9954747Z 	at com.purbon.kafka.topology.schemas.SchemaRegistryManager.register(SchemaRegistryManager.java:45)
2020-09-29T15:59:38.9955290Z 	at com.purbon.kafka.topology.actions.topics.SyncTopicAction.syncTopic(SyncTopicAction.java:60)
2020-09-29T15:59:38.9955758Z 	at com.purbon.kafka.topology.actions.topics.SyncTopicAction.run(SyncTopicAction.java:41)
2020-09-29T15:59:38.9956174Z 	at com.purbon.kafka.topology.ExecutionPlan.execute(ExecutionPlan.java:66)
2020-09-29T15:59:38.9956540Z 	at com.purbon.kafka.topology.ExecutionPlan.run(ExecutionPlan.java:50)
2020-09-29T15:59:38.9956926Z 	at com.purbon.kafka.topology.KafkaTopologyBuilder.run(KafkaTopologyBuilder.java:105)
2020-09-29T15:59:38.9957340Z 	at com.purbon.kafka.topology.KafkaTopologyBuilder.run(KafkaTopologyBuilder.java:116)
2020-09-29T15:59:38.9957948Z 	at com.purbon.kafka.topology.BuilderCLI.processTopology(BuilderCLI.java:178)
2020-09-29T15:59:38.9958308Z 	at com.purbon.kafka.topology.BuilderCLI.run(BuilderCLI.java:131)
2020-09-29T15:59:38.9958649Z 	at com.purbon.kafka.topology.BuilderCLI.main(BuilderCLI.java:120)
2020-09-29T15:59:38.9959622Z Caused by: com.purbon.kafka.topology.schemas.SchemaRegistryManager$SchemaRegistryManagerException: Failed to parse the schema file poc_telemetry/poc_telemetry/schemas/bar-key.avsc
2020-09-29T15:59:38.9960217Z 	at com.purbon.kafka.topology.schemas.SchemaRegistryManager.register(SchemaRegistryManager.java:54)
2020-09-29T15:59:38.9960695Z 	at com.purbon.kafka.topology.schemas.SchemaRegistryManager.register(SchemaRegistryManager.java:43)
2020-09-29T15:59:38.9960994Z 	... 9 more
2020-09-29T15:59:38.9961716Z Caused by: com.purbon.kafka.topology.schemas.SchemaRegistryManager$SchemaRegistryManagerException: Failed to register the schema for subject 'context.company.env.source.projectB.bar.avro' of type 'AVRO'
2020-09-29T15:59:38.9962346Z 	at com.purbon.kafka.topology.schemas.SchemaRegistryManager.register(SchemaRegistryManager.java:86)
2020-09-29T15:59:38.9962814Z 	at com.purbon.kafka.topology.schemas.SchemaRegistryManager.register(SchemaRegistryManager.java:52)
2020-09-29T15:59:38.9963113Z 	... 10 more
2020-09-29T15:59:38.9963460Z Caused by: io.confluent.kafka.schemaregistry.client.rest.exceptions.RestClientException: Unauthorized; error code: 401
2020-09-29T15:59:38.9964088Z 	at io.confluent.kafka.schemaregistry.client.rest.RestService.sendHttpRequest(RestService.java:293)
2020-09-29T15:59:38.9964544Z 	at io.confluent.kafka.schemaregistry.client.rest.RestService.httpRequest(RestService.java:352)
2020-09-29T15:59:38.9965013Z 	at io.confluent.kafka.schemaregistry.client.rest.RestService.registerSchema(RestService.java:495)
2020-09-29T15:59:38.9965473Z 	at io.confluent.kafka.schemaregistry.client.rest.RestService.registerSchema(RestService.java:486)
2020-09-29T15:59:38.9965936Z 	at io.confluent.kafka.schemaregistry.client.rest.RestService.registerSchema(RestService.java:459)
2020-09-29T15:59:39.0016821Z 	at io.confluent.kafka.schemaregistry.client.CachedSchemaRegistryClient.registerAndGetId(CachedSchemaRegistryClient.java:206)
2020-09-29T15:59:39.0017414Z 	at io.confluent.kafka.schemaregistry.client.CachedSchemaRegistryClient.register(CachedSchemaRegistryClient.java:268)
2020-09-29T15:59:39.0017966Z 	at io.confluent.kafka.schemaregistry.client.CachedSchemaRegistryClient.register(CachedSchemaRegistryClient.java:244)
2020-09-29T15:59:39.0018488Z 	at com.purbon.kafka.topology.schemas.SchemaRegistryManager.register(SchemaRegistryManager.java:80)
2020-09-29T15:59:39.0018791Z 	... 11 more`

--- a/.gitignore
+++ b/.gitignore
@@ -9,3 +9,4 @@ _build
 
 dependency-reduced-pom.xml
 server-api/logs/
+example/topology-builder-with-schema-cloud.properties

--- a/changelog.md
+++ b/changelog.md
@@ -21,6 +21,7 @@ it will have to be properly configured.
 * Fix bug on removal of acls. This solve issue #72.
 * Fix bug with schema registration and file path reference,  now by default all are relative to the main directory where the topology file is staying.
 * Improved delete and general management of acls logic, only create what is necessary, only delete what is mandatory.
+* Fixed a problem with the schema registry client that got no configs, so it would not be possible to use with security contrain systems such as confluent cloud.
 
 v1.0.0-rc1:
 * Add support for platform wide acls for control center in teh topology description file.

--- a/scripts/run-docker-run-cmd.sh
+++ b/scripts/run-docker-run-cmd.sh
@@ -1,0 +1,10 @@
+#!/usr/bin/env bash
+
+docker run -t -i \
+      -v /Users/pere/work/kafka-topology-builder/example:/example \
+      purbon/kafka-topology-builder:srlatest \
+      kafka-topology-builder.sh \
+      --brokers pkc-4ygn6.europe-west3.gcp.confluent.cloud:9092 \
+      --clientConfig /example/topology-builder-with-schema-cloud.properties \
+      --topology /example/descriptor.yaml -quiet
+

--- a/src/main/java/com/purbon/kafka/topology/KafkaTopologyBuilder.java
+++ b/src/main/java/com/purbon/kafka/topology/KafkaTopologyBuilder.java
@@ -12,6 +12,7 @@ import com.purbon.kafka.topology.model.Topology;
 import com.purbon.kafka.topology.schemas.SchemaRegistryManager;
 import io.confluent.kafka.schemaregistry.client.CachedSchemaRegistryClient;
 import io.confluent.kafka.schemaregistry.client.SchemaRegistryClient;
+import io.confluent.kafka.schemaregistry.client.rest.RestService;
 import java.io.IOException;
 import java.io.InputStream;
 import java.io.PrintStream;
@@ -70,8 +71,11 @@ public class KafkaTopologyBuilder implements AutoCloseable {
     AccessControlManager accessControlManager =
         new AccessControlManager(accessControlProvider, bindingsBuilderProvider, config);
 
+    RestService restService = new RestService(config.getConfluentSchemaRegistryUrl());
+    Map<String, ?> schemaRegistryConfig = config.asMap("schema.registry");
     SchemaRegistryClient schemaRegistryClient =
-        new CachedSchemaRegistryClient(config.getConfluentSchemaRegistryUrl(), 10);
+        new CachedSchemaRegistryClient(
+            restService, 10, schemaRegistryConfig.isEmpty() ? null : schemaRegistryConfig);
     SchemaRegistryManager schemaRegistryManager =
         new SchemaRegistryManager(schemaRegistryClient, topologyFileOrDir);
 

--- a/src/main/java/com/purbon/kafka/topology/KafkaTopologyBuilder.java
+++ b/src/main/java/com/purbon/kafka/topology/KafkaTopologyBuilder.java
@@ -72,7 +72,7 @@ public class KafkaTopologyBuilder implements AutoCloseable {
         new AccessControlManager(accessControlProvider, bindingsBuilderProvider, config);
 
     RestService restService = new RestService(config.getConfluentSchemaRegistryUrl());
-    Map<String, ?> schemaRegistryConfig = config.asMap("schema.registry");
+    Map<String, ?> schemaRegistryConfig = config.asMap();
     SchemaRegistryClient schemaRegistryClient =
         new CachedSchemaRegistryClient(
             restService, 10, schemaRegistryConfig.isEmpty() ? null : schemaRegistryConfig);

--- a/src/main/java/com/purbon/kafka/topology/TopologyBuilderConfig.java
+++ b/src/main/java/com/purbon/kafka/topology/TopologyBuilderConfig.java
@@ -80,6 +80,14 @@ public class TopologyBuilderConfig {
     this.properties = properties;
   }
 
+  public Map<String, ?> asMap(String filter) {
+    Map<String, Object> map = new HashMap<>();
+    properties.keySet().stream()
+        .filter(o -> String.valueOf(o).startsWith(filter))
+        .forEach(key -> map.put(String.valueOf(key), properties.get(key)));
+    return map;
+  }
+
   public void validateWith(Topology topology) throws ConfigurationException {
 
     validateGeneralConfiguration(topology);

--- a/src/main/java/com/purbon/kafka/topology/TopologyBuilderConfig.java
+++ b/src/main/java/com/purbon/kafka/topology/TopologyBuilderConfig.java
@@ -52,7 +52,7 @@ public class TopologyBuilderConfig {
   public static final String MDS_KC_CLUSTER_ID_CONFIG =
       "topology.builder.mds.kafka.connect.cluster.id";
 
-  public static final String CONFLUENT_SCHEMA_REGISTRY_URL_CONFIG = "confluent.schema.registry.url";
+  public static final String CONFLUENT_SCHEMA_REGISTRY_URL_CONFIG = "schema.registry.url";
   public static final String CONFLUENT_SCHEMA_REGISTRY_URL_DEFAULT = "mock://";
 
   public static final String CONFLUENT_MONITORING_TOPIC_CONFIG = "confluent.monitoring.topic";
@@ -80,10 +80,14 @@ public class TopologyBuilderConfig {
     this.properties = properties;
   }
 
+  public Map<String, ?> asMap() {
+    return asMap("");
+  }
+
   public Map<String, ?> asMap(String filter) {
     Map<String, Object> map = new HashMap<>();
     properties.keySet().stream()
-        .filter(o -> String.valueOf(o).startsWith(filter))
+        .filter(o -> filter.isEmpty() || String.valueOf(o).startsWith(filter))
         .forEach(key -> map.put(String.valueOf(key), properties.get(key)));
     return map;
   }

--- a/src/main/java/com/purbon/kafka/topology/actions/topics/SyncTopicAction.java
+++ b/src/main/java/com/purbon/kafka/topology/actions/topics/SyncTopicAction.java
@@ -57,11 +57,11 @@ public class SyncTopicAction extends BaseAction {
       final TopicSchemas schemas = topic.getSchemas();
 
       if (StringUtils.isNotBlank(schemas.getKeySchemaFile())) {
-        schemaRegistryManager.register(fullTopicName, schemas.getKeySchemaFile());
+        schemaRegistryManager.register(fullTopicName + "-key", schemas.getKeySchemaFile());
       }
 
       if (StringUtils.isNotBlank(schemas.getValueSchemaFile())) {
-        schemaRegistryManager.register(fullTopicName, schemas.getValueSchemaFile());
+        schemaRegistryManager.register(fullTopicName + "-value", schemas.getValueSchemaFile());
       }
     }
   }

--- a/src/test/resources/client-config-redis.properties
+++ b/src/test/resources/client-config-redis.properties
@@ -3,7 +3,7 @@ sasl.mechanism=PLAIN
 sasl.jaas.config=org.apache.kafka.common.security.plain.PlainLoginModule required \
   username="kafka" \
   password="kafka";
-confluent.schema.registry.url="http://localhost:8082"
+schema.registry.url="http://localhost:8082"
 
 topology.builder.state.processor.class=com.purbon.kafka.topology.clusterstate.RedisStateProcessor
 topology.builder.redis.host=foo

--- a/src/test/resources/client-config.properties
+++ b/src/test/resources/client-config.properties
@@ -3,4 +3,4 @@ sasl.mechanism=PLAIN
 sasl.jaas.config=org.apache.kafka.common.security.plain.PlainLoginModule required \
   username="kafka" \
   password="kafka";
-confluent.schema.registry.url="http://localhost:8082"
+schema.registry.url="http://localhost:8082"

